bluetooth/src/commonMain/kotlin/Bluetooth.kt

```diff
@@ -0,0 +1,296 @@
+    }
+
+    private val scanMode = MutableStateFlow<ScanMode>(ScanMode.Stopped)
+
+    fun collectRepo(it:Any) {
+        println(it)
+
+    }
+
+    fun devices(): Flow<List<Device>> {
+
+        launch {
+            scanningStateRepo.collect {
+                collectRepo(it)
+            }
+        }
+
+            return combine(scanningStateRepo, scanMode) { scanState, scanMode ->
+                when (scanState) {
+                    is ScanningState.Initialized.Enabled.Idle -> {
+                        when(scanMode) {
+                            is ScanMode.Stopped -> emptyList()
+                            is ScanMode.Scan -> {
+                                scanningStateRepo.takeAndChangeState(remainIfStateNot = ScanningState.Initialized.Enabled.Idle::class) {
+                                    it.startScanning(scanMode.filter)
+                                }
+                                if (scanState.discovered.filter == scanMode.filter)
+                                    scanState.discovered.devices
+                                else
+                                    emptyList()
+
+                            }
+                        }
+                    }
+                    is ScanningState.Initialized.Enabled.Scanning -> {
+
+                        when (scanMode) {
+                            is ScanMode.Scan -> {
+                                d("devices: ${scanState.discovered}")
+                                if (scanState.discovered.filter == scanMode.filter)
+                                    scanState.discovered.devices
+                                else {
+                                    scanningStateRepo.takeAndChangeState(ScanningState.Initialized.Enabled.Scanning::class) { state ->
+                                        state.stopScanning
+                                    }
+                                    emptyList()
+                                }
+                            }
+                            else -> {
+                                scanningStateRepo.takeAndChangeState(ScanningState.Initialized.Enabled.Scanning::class) { state ->
+                                    state.stopScanning
+                                }
+                                scanState.discovered.devices
+                            }
+                        }
+                    }
+
+                    is ScanningState.Initialized.NoBluetooth -> {
+                        emptyList()
+                    }
+
+                    is ScanningState.NotInitialized -> {
+                        scanningStateRepo.takeAndChangeState(ScanningState.NotInitialized::class) { state ->
+                            state.initialize(scanningStateRepo)
+                        }
+                        emptyList()
+                    }
+
+                }
+            }
+    }
+
+    fun startScanning(filter: Set<UUID> = emptySet()) {
```

Comment:
This UUID object has no constructor, so it's not really useful from common code. Can we provide default constructor to create UUID object from given string?

Reply:
I can add a method here: 

https://github.com/splendo/kaluga/blob/9c116031f4be67c96308a82ab9eab8c0344cea8b/bluetooth/src/commonMain/kotlin/UUID.kt#L22